clients/azuredevopsrepo/client.go

```diff
@@ -0,0 +1,216 @@
+}
+
+func (client *Client) GetCreatedAt() (time.Time, error) {
+	return time.Time{}, nil
+}
+
+func (client *Client) GetDefaultBranchName() (string, error) {
+	if len(client.repourl.defaultBranch) > 0 {
+		return client.repourl.defaultBranch, nil
+	}
+
+	return "", fmt.Errorf("%s", "default branch name is empty")
+}
+
+func (client *Client) GetDefaultBranch() (*clients.BranchRef, error) {
+	return client.branches.getDefaultBranch()
+}
+
+func (client *Client) GetOrgRepoClient(context.Context) (clients.RepoClient, error) {
+	return nil, fmt.Errorf("GetOrgRepoClient (GitLab): %w", clients.ErrUnsupportedFeature)
+}
+
+func (client *Client) ListCommits() ([]clients.Commit, error) {
+	return client.commits.listCommits()
+}
+
+func (client *Client) ListIssues() ([]clients.Issue, error) {
+	return nil, nil
+}
+
+func (client *Client) ListLicenses() ([]clients.License, error) {
+	return nil, nil
+}
+
+func (client *Client) ListReleases() ([]clients.Release, error) {
+	return nil, nil
+}
+
+func (client *Client) ListContributors() ([]clients.User, error) {
+	return nil, nil
+}
+
+func (client *Client) ListSuccessfulWorkflowRuns(filename string) ([]clients.WorkflowRun, error) {
+	return nil, nil
+}
+
+func (client *Client) ListCheckRunsForRef(ref string) ([]clients.CheckRun, error) {
+	return nil, nil
+}
+
+func (client *Client) ListStatuses(ref string) ([]clients.Status, error) {
+	return nil, nil
+}
+
+func (client *Client) ListWebhooks() ([]clients.Webhook, error) {
+	return nil, nil
+}
+
+func (client *Client) ListProgrammingLanguages() ([]clients.Language, error) {
+	return nil, nil
+}
+
+func (client *Client) Search(request clients.SearchRequest) (clients.SearchResponse, error) {
+	return clients.SearchResponse{}, nil
+}
+
+func (client *Client) SearchCommits(request clients.SearchCommitsOptions) ([]clients.Commit, error) {
+	return nil, nil
+}
+
+func (client *Client) Close() error {
+	return nil
+}
+
+func CreateAzureDevOpsClient(ctx context.Context, repo clients.Repo) (*Client, error) {
+	token := os.Getenv("AZURE_DEVOPS_AUTH_TOKEN")
+	return CreateAzureDevOpsClientWithToken(ctx, token, repo)
+}
+
+func CreateAzureDevOpsClientWithToken(ctx context.Context, token string, repo clients.Repo) (*Client, error) {
+	// https://dev.azure.com/<org>
+	url := "https://" + repo.Host() + "/" + strings.Split(repo.Path(), "/")[0]
+	connection := azuredevops.NewPatConnection(url, token)
+
+	gitClient, err := git.NewClient(ctx, connection)
+	if err != nil {
+		return nil, fmt.Errorf("could not create azure devops git client with error: %w", err)
+	}
+
+	return &Client{
+		ctx:        ctx,
+		azdoClient: gitClient,
+		branches: &branchesHandler{
+			gitClient: gitClient,
+		},
+		commits: &commitsHandler{
+			gitClient: gitClient,
+		},
+	}, nil
+}
+
+func CreateOssFuzzRepoClient(ctx context.Context, logger *log.Logger) (clients.RepoClient, error) {
+	return nil, fmt.Errorf("%w, oss fuzz currently only supported for github repos", clients.ErrUnsupportedFeature)
```

Comment:
This isn't part of the `RepoClient` interface, let's just delete this. 

Plus oss fuzz supports arbitrary git hosts

Reply:
Removed.